tests/examples/shims/client.lua

```diff
@@ -0,0 +1,202 @@
+local gears_obj = require("gears.object")
+
+local clients = {}
+
+--TODO merge all other implementations
+local function convert_singleton(obj)
+    obj._connect_signal = obj.connect_signal
+    function obj.connect_signal(name, func)
+        return obj._connect_signal(obj, name, func)
+    end
+    obj.weak_connect_signal = obj.connect_signal
```

Comment:
The C-API doesn't have weak signal connections, so I think this part can be dropped.